pkg/cmd/controller/csr.go

```diff
@@ -10,37 +11,66 @@ import (
 )
 
 const (
-	controllerName                    = "csr-approver-controller"
-	monitoringServiceAccountNamespace = "openshift-monitoring"
-	monitoringServiceAccountName      = "cluster-monitoring-operator"
-	monitoringCertificateSubject      = "CN=system:serviceaccount:openshift-monitoring:prometheus-k8s"
-	monitoringLabelKey                = "metrics.openshift.io/csr.subject"
-	monitoringLabelValue              = "prometheus"
+	controllerNamePrefix                  = "csr-approver-controller"
+	monitoringNamespace                   = "openshift-monitoring"
+	monitoringRequesterServiceAccountName = "cluster-monitoring-operator"
+	monitoringSubjectNameLabelKey         = "metrics.openshift.io/csr.subject"
 )
 
+type monitoringCSR struct {
+	subject     string
+	certSubject string
+	selector    labels.Selector
+}
+
+func newMonitoringCSR(subjectName, subjectServiceAcount string) (*monitoringCSR, error) {
+	labelsRequirement, err := labels.NewRequirement(monitoringSubjectNameLabelKey, selection.Equals, []string{subjectName})
+	if err != nil {
+		return nil, err
+	}
+	return &monitoringCSR{
+		subject:     subjectName,
+		selector:    labels.NewSelector().Add(*labelsRequirement),
+		certSubject: fmt.Sprintf("CN=system:serviceaccount:%s:%s", monitoringNamespace, subjectServiceAcount),
+	}, nil
+}
+
+// RunCSRApproverController approves the CSRs for prometheus and metrics-server client certs that are used for authn against various /metrics endpoints.
 func RunCSRApproverController(ctx context.Context, controllerCtx *EnhancedControllerContext) (bool, error) {
 	kubeClient, err := controllerCtx.ClientBuilder.Client(infraClusterCSRApproverControllerServiceAccountName)
 	if err != nil {
 		return true, err
 	}
 
-	selector := labels.NewSelector()
-	labelsRequirement, err := labels.NewRequirement(monitoringLabelKey, selection.Equals, []string{monitoringLabelValue})
```

Comment:
Perhaps we should use `labelsRequirement, err := labels.NewRequirement(monitoringLabelKey, selection.In, []string{prometheusValue,metricsServerValue})` instead of two controllers?

Reply:
If you really mean "controllers":
We'll need two controllers because each corresponding CSRApprover can only handle one SA.
(same as https://github.com/openshift/cluster-policy-controller/pull/148#discussion_r1580831905)

If you mean "selectors":
I've already try that https://github.com/openshift/cluster-policy-controller/compare/bee77685938fb3fd9f78444cd8fbd2c282392f21..a881be203921d0d69023d97878cc91f85ccf409d and the controllers will step on each other’s toes: One will deny the requests of the other

Reply:
```
status:
  conditions:
  - lastTransitionTime: "2024-04-26T08:16:06Z"
    lastUpdateTime: "2024-04-26T08:16:06Z"
    message: expected the CSR's subject to be "CN=system:serviceaccount:openshift-monitoring:metrics-server",
      but it is "CN=system:serviceaccount:openshift-monitoring:prometheus-k8s"
    reason: CSRDenied
    status: "True"
    type: Denied
```

Reply:
Oh, right, serviceaccounts are different too.

I think ideally we'll want to rework this:
* `NewMultipleServiceAccountApprover` function accepting an array of serviceaccounts
* selector with `In`
* rework to use single controller

WDYT?